server/src/main/scala/drt/server/feeds/edi/EdiFeed.scala

```diff
@@ -0,0 +1,154 @@
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+trait EdiFeedJsonSupport extends DefaultJsonProtocol with SprayJsonSupport {
+
+  implicit val ediFlightDetailsFormat: RootJsonFormat[EdiFlightDetails] = jsonFormat22(EdiFlightDetails)
+
+  def unMarshalResponseToEdiFlightDetails(httpResponse: HttpResponse)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Future[List[EdiFlightDetails]] = {
+    httpResponse.entity.dataBytes.runReduce(_ ++ _)
+      .flatMap(d => Unmarshal(d.utf8String).to[List[EdiFlightDetails]])
+  }
+}
+
+class EdiFeed(ediClient: EdiClient) extends EdiFeedJsonSupport {
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def makeRequestAndFeedResponseToArrivalSource(startData: String, endDate: String, feedSource: FeedSource)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Future[ArrivalsFeedResponse] = {
+    log.info(s"$feedSource Edi feed api call at ${DateTime.now()} from $startData and $endDate ")
+    ediClient.makeRequest(startData, endDate).flatMap { response =>
+      response.status match {
+        case OK => unMarshalResponseToEdiFlightDetails(response).map { flights =>
+          log.info(s"Edi feed status ${response.status} with api call for ${flights.size} flights")
+          ArrivalsFeedSuccess(Flights(ediFlightDetailsToArrival(flights, feedSource)))
+        }
+        case _ => log.warn(s"Edi feed status ${response.status} while api call with response ${response.entity}")
+          Future.successful(ArrivalsFeedFailure(s"Response with status ${response.status} from edi ${response.entity}"))
+      }
+    }.recover { case t =>
+      log.error(s"Edi feed error while api call ${t.getMessage}")
+      ArrivalsFeedFailure(t.getMessage)
+    }
+  }
+
+  def ediFeedPollingSource(interval: FiniteDuration, feedSource: FeedSource)(implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext): Source[ArrivalsFeedResponse, Cancellable] = {
+    var startForecastDays = 2
+    var endForecastDays = 30
+
+    Source.tick(1 seconds, interval, NotUsed)
+      .mapAsync(1) { _ =>
+        feedSource match {
+          case LiveFeedSource =>
+            val currentDate = SDate.yyyyMmDdForZone(SDate.now(), DateTimeZone.UTC)
+            val endDate = SDate.yyyyMmDdForZone(JodaSDate(new DateTime(DateTimeZone.UTC).plusDays(2)), DateTimeZone.UTC)
+            makeRequestAndFeedResponseToArrivalSource(currentDate, endDate, feedSource)
+
+          case ForecastFeedSource =>
+            val startDate = SDate.yyyyMmDdForZone(JodaSDate(new DateTime(DateTimeZone.UTC).plusDays(startForecastDays)), DateTimeZone.UTC)
+            val endDate = SDate.yyyyMmDdForZone(JodaSDate(new DateTime(DateTimeZone.UTC).plusDays(endForecastDays)), DateTimeZone.UTC)
+            if (endForecastDays >= 180) {
```

Comment:
This looks very over complicated and difficult to read. Why are we assigning `endForecastDays` to `startForecastDays`?

We shouldn't need any `var`s. Just either pass the number of days when creating the feed or hard code it if we're unlikely to ever want to change it.